core/admin/mailu/internal/views/postfix.py

```diff
@@ -18,37 +18,32 @@ def postfix_mailbox_map(email):
 
 @internal.route("/postfix/alias/<alias>")
 def postfix_alias_map(alias):
-    localpart, domain = alias.split('@', 1) if '@' in alias else (None, alias)
-    alternative = models.Alternative.query.get(domain)
-    if alternative:
-        domain = alternative.domain_name
-    email = '{}@{}'.format(localpart, domain)
+    localpart, domain_name = models.Email.resolve_domain(alias)
     if localpart is None:
-        return flask.jsonify(domain)
-    else:
-        alias_obj = models.Alias.resolve(localpart, domain)
-        if alias_obj:
-            return flask.jsonify(",".join(alias_obj.destination))
-        user_obj = models.User.query.get(email)
-        if user_obj:
-            return flask.jsonify(user_obj.destination)
-        return flask.abort(404)
+        return flask.jsonify(domain_name)
+    destination = models.Email.resolve_destination(localpart, domain_name)
+    return flask.jsonify(",".join(destination)) if destination else flask.abort(404)
 
 
 @internal.route("/postfix/transport/<email>")
 def postfix_transport(email):
-    localpart, domain = email.split('@', 1) if '@' in email else (None, email)
-    relay = models.Relay.query.get(domain) or flask.abort(404)
+    localpart, domain_name = models.Email.resolve_domain(email)
```

Comment:
At one point it seems and asterisk (`*`) is passed here, which fails against the IDNA type.

```
admin_1      | The above exception was the direct cause of the following exception:
admin_1      | 
admin_1      | Traceback (most recent call last):
admin_1      |   File "/usr/lib/python3.6/site-packages/flask/app.py", line 1982, in wsgi_app
admin_1      |     response = self.full_dispatch_request()
admin_1      |   File "/usr/lib/python3.6/site-packages/flask/app.py", line 1614, in full_dispatch_request
admin_1      |     rv = self.handle_user_exception(e)
admin_1      |   File "/usr/lib/python3.6/site-packages/flask/app.py", line 1517, in handle_user_exception
admin_1      |     reraise(exc_type, exc_value, tb)
admin_1      |   File "/usr/lib/python3.6/site-packages/flask/_compat.py", line 33, in reraise
admin_1      |     raise value
admin_1      |   File "/usr/lib/python3.6/site-packages/flask/app.py", line 1612, in full_dispatch_request
admin_1      |     rv = self.dispatch_request()
admin_1      |   File "/usr/lib/python3.6/site-packages/flask/app.py", line 1598, in dispatch_request
admin_1      |     return self.view_functions[rule.endpoint](**req.view_args)
admin_1      |   File "/app/mailu/internal/views/postfix.py", line 30, in postfix_transport
admin_1      |     localpart, domain_name = models.Email.resolve_domain(email)
admin_1      |   File "/app/mailu/models.py", line 231, in resolve_domain
admin_1      |     alternative = Alternative.query.get(domain_name)
admin_1      |   File "/usr/lib/python3.6/site-packages/sqlalchemy/orm/query.py", line 882, in get
admin_1      |     ident, loading.load_on_ident)
admin_1      |   File "/usr/lib/python3.6/site-packages/sqlalchemy/orm/query.py", line 916, in _get_impl
admin_1      |     return fallback_fn(self, key)
admin_1      |   File "/usr/lib/python3.6/site-packages/sqlalchemy/orm/loading.py", line 232, in load_on_ident
admin_1      |     return q.one()
admin_1      |   File "/usr/lib/python3.6/site-packages/sqlalchemy/orm/query.py", line 2848, in one
admin_1      |     ret = self.one_or_none()
admin_1      |   File "/usr/lib/python3.6/site-packages/sqlalchemy/orm/query.py", line 2818, in one_or_none
admin_1      |     ret = list(self)
admin_1      |   File "/usr/lib/python3.6/site-packages/sqlalchemy/orm/query.py", line 2889, in __iter__
admin_1      |     return self._execute_and_instances(context)
admin_1      |   File "/usr/lib/python3.6/site-packages/sqlalchemy/orm/query.py", line 2912, in _execute_and_instances
admin_1      |     result = conn.execute(querycontext.statement, self._params)
admin_1      |   File "/usr/lib/python3.6/site-packages/sqlalchemy/engine/base.py", line 948, in execute
admin_1      |     return meth(self, multiparams, params)
admin_1      |   File "/usr/lib/python3.6/site-packages/sqlalchemy/sql/elements.py", line 269, in _execute_on_connection
admin_1      |     return connection._execute_clauseelement(self, multiparams, params)
admin_1      |   File "/usr/lib/python3.6/site-packages/sqlalchemy/engine/base.py", line 1060, in _execute_clauseelement
admin_1      |     compiled_sql, distilled_params
admin_1      |   File "/usr/lib/python3.6/site-packages/sqlalchemy/engine/base.py", line 1132, in _execute_context
admin_1      |     None, None)
admin_1      |   File "/usr/lib/python3.6/site-packages/sqlalchemy/engine/base.py", line 1413, in _handle_dbapi_exception
admin_1      |     exc_info
admin_1      |   File "/usr/lib/python3.6/site-packages/sqlalchemy/util/compat.py", line 203, in raise_from_cause
admin_1      |     reraise(type(exception), exception, tb=exc_tb, cause=cause)
admin_1      |   File "/usr/lib/python3.6/site-packages/sqlalchemy/util/compat.py", line 186, in reraise
admin_1      |     raise value.with_traceback(tb)
admin_1      |   File "/usr/lib/python3.6/site-packages/sqlalchemy/engine/base.py", line 1127, in _execute_context
admin_1      |     context = constructor(dialect, self, conn, *args)
admin_1      |   File "/usr/lib/python3.6/site-packages/sqlalchemy/engine/default.py", line 667, in _init_compiled
admin_1      |     param.append(processors[key](compiled_params[key]))
admin_1      |   File "/usr/lib/python3.6/site-packages/sqlalchemy/sql/type_api.py", line 1160, in process
admin_1      |     return process_param(value, dialect)
admin_1      |   File "/app/mailu/models.py", line 25, in process_bind_param
admin_1      |     return idna.encode(value).decode("ascii")
admin_1      |   File "/usr/lib/python3.6/site-packages/idna/core.py", line 355, in encode
admin_1      |     result.append(alabel(label))
admin_1      |   File "/usr/lib/python3.6/site-packages/idna/core.py", line 265, in alabel
admin_1      |     raise IDNAError('The label {0} is not a valid A-label'.format(label))
admin_1      | sqlalchemy.exc.StatementError: (idna.core.IDNAError) The label b'*' is not a valid A-label [SQL: 'SELECT alternative.created_at AS alternative_created_at, alternative.updated_at AS alternative_updated_at, alternative.comment AS alternative_comment, alternative.name AS alternative_name, alternative.domain_name AS alternative_domain_name \nFROM alternative \nWHERE alternative.name = ?'] [parameters: [{'%(140545964121000 param)s': '*'}]]
```

Reply:
So on master the asterisk was always there, just it seems never to reach the type-check.

```
192.168.203.2 - - [04/Dec/2018:13:07:22 +0000] "GET /internal/postfix/transport/* HTTP/1.1" 404 233 "-" "Python/3.6 aiohttp/3.4.4"
```
It seems to always result in a 404. So if we catch early in `postfix_transport(email)` and return a 404, it should be ok.